src/core/mac/sub_mac.hpp

```diff
@@ -507,6 +507,7 @@ class SubMac : public InstanceLocator, private NonCopyable
     static void HandleCslTimer(Timer &aTimer);
     void        HandleCslTimer(void);
     void        GetCslWindowEdges(uint32_t &aAhead, uint32_t &aAfter);
```

Comment:
I think the problem is caused by the inconsistent usage of `GetCslWindowEdges()`.

If `RadioSupportsReceiveTiming()`, it's expected to calculate the timing after a period. However, if not `RadioSupportsReceiveTiming()`, it's expected to calculate the timing of the current period.

Perhaps, we could add a parameter for this method specifying the sample time in local time. In the implementation, it doesn't calculate the timing according to the current time, but the sample time.

And in the `HandleCslTimer()`, feeding different sample time based on `RadioSupportsReceiveTiming()`.
```suggestion
    void        GetCslWindowEdges(uint32_t aSampleTime, uint32_t &aAhead, uint32_t &aAfter);
```

Reply:
Thanks @bukepo , the crux of this PR indeed lies in the fact that `GetCslWindowEdges()` isn't used consistently, which is why we have introduced a drift value. 

Regardless of whether `RadioSupportsReceiveTiming()` is true or not, there are always 2 uses of `GetCslWindowEdges()` within a single CSL cycle:
- schedule next trigger of CSL timer (this PR aims to make changes to this)
- schedule an Rx window (given start time + duration) for the current trigger of CSL timer (there are no changes to this in this PR)

Adding another parameter to the method would be helpful for readability if (not `RadioSupportsReceiveTiming()`), but would require more edits if `RadioSupportsReceiveTiming()` because the timer is only triggered once per cycle (such as changing the number of parameters for `HandleCslReceiveAt()`).

Either way, the modifications should result in the same outcome as the edits in this PR, because we are only aiming to change the scheduling of next trigger of CSL timer.

Reply:
Yes, it should result the same calculation. However, every time I read the code here, it takes extra time to understand the intention. I'd suggest we choose a fix that has better readability for maintenance. Thoughts?